### DISCUSSION TITLE
`--version` to show applied patches

### DISCUSSION THREAD

**Discussion by vszakats** [curl/curl#13869] [Ideas]
Curl has a feature to pass applied security fixes as a string and display it in `--version` output. It's a macro called `CURL_PATCHSTAMP`:
https://github.com/curl/curl/blob/4edbd5226792c6175f0abdacfb70d66984e019be/src/tool_help.c#L180-L183

To make patches applied to curl-for-win builds visible, I tried using it (https://github.com/curl/curl-for-win/commit/7cd6bfa8b0e242dd3838e53fcc8edebfc4c83a15), and this is how it appears:
```
curl 8.8.0 (x86_64-alpine-linux-musl) libcurl/8.8.0 LibreSSL/3.9.2 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libpsl/0.21.5 libssh2/1.11.0 nghttp2/1.62.1 ngtcp2/1.5.0 nghttp3/1.3.0
Release-Date: 2024-05-22, security patched: 582743f2,c61f7538
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 HTTPS-proxy IPv6 Largefile libz NTLM PSL SSL threadsafe UnixSockets zstd
```
Ref: https://github.com/curl/curl-for-win/actions/runs/9356737626/job/25754883996#step:3:6102

But, "security patches:" is misleading here, because it's _sometimes_ security related but often it is not, just "important enough" to include between stable releases.

Would it make sense to drop "security" from the output, or introduce another option to indicate non-security patches?

**Comment by bagder** [comment]
The idea for that define was to provide a date, to indicate that there are details in the build that are newer than the (original) release date. I believe that is a better message rather than a set of hashes that are completely meaningless to most users. Even if contains less information.

**Comment by vszakats** [comment]
Thanks for the insight.

A date (or more dates) come(s) short to actually identify what was applied.
I could probably roll with it, but it's somewhat clunky to extract them from the 
patch file and convert/sort to a single value. 

So, I'm thinking to include a permalink to the actual patch file (it contains all 
the patches for that build). Less clunky to implement, and it gives all the info
in a single URL:
```
curl 8.8.0 (x86_64-alpine-linux-musl) libcurl/8.8.0 LibreSSL/3.9.2 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libpsl/0.21.5 libssh2/1.11.0 nghttp2/1.62.1 ngtcp2/1.5.0 nghttp3/1.3.0
Release-Date: 2024-05-22, security patched: https://github.com/curl/curl-for-win/blob/8bb77c8/curl.test.patch
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 HTTPS-proxy IPv6 Largefile libz NTLM PSL SSL threadsafe UnixSockets zstd
```
Ref: https://github.com/curl/curl-for-win/actions/runs/9370491111/job/25797582780#step:3:6104
